### PR TITLE
626 platform report

### PR DESCRIPTION
**Pull request by summer-cook**
# Story
Builds the counter compliant platform report

# Related
- #626 

# Expected Behavior Before Changes
- JSON returned a simple key/value pair

# Expected Behavior After Changes
- [x] JSON for the platform report endpoint returns data for total investigations & total requests
- [x] Data can be filtered by begin date and end date
- [x] Data can be filtered by data_type (aligns with resource_type in hyku)

# Screenshots / Video

<details>
<summary>JSON</summary>
image

</details>

### CODE REVIEW

File: spec/models/platform_report_spec.rb
Comment (labradford) [inline spec/models/platform_report_spec.rb:1]: we should probably move this into a reports folder to follow the same path as the model file

Comment (summer-cook) [reply spec/models/platform_report_spec.rb:1]: good call

File: app/models/reports/platform_report.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      @created = created
+      @account = account
+
+      # We want to limit the available attributes to be a subset of the given attributes; the `&` is
+      # the intersection of the two arrays.
+      @attributes_to_show = params.fetch(:attributes_to_show, ["Access_Method"]) & ALLOWED_REPORT_ATTRIBUTES_TO_SHOW
+
+      # Because we're receiving user input that is likely strings, we need to do some coercion.
+      @begin_date = self.class.coerce_to_date(params.fetch(:begin_date))
+      @end_date = self.class.coerce_to_date(params.fetch(:end_date))
```
Comment (labradford) [inline app/models/reports/platform_report.rb:43]: The end data will always be the last day of the month, so coerce_to_date should take that into account.

Comment (labradford) [reply app/models/reports/platform_report.rb:43]: probably can address this in a future PR

Comment (summer-cook) [reply app/models/reports/platform_report.rb:43]: I think based on how it is set up, in that case they would query with begin date as the month they want, end date as the following month, or first day of the following month: for instance, begin_date=2023-06, end_date=2023-07 would be June 2023. Does that work?

Comment (labradford) [reply app/models/reports/platform_report.rb:43]: The specification is begin date is first of the month, end date is the last day of the month. We can figure it out later

**Review by labradford — APPROVED** [review]

File: app/models/reports/platform_report.rb (hunk 1 above)
Comment (alishaevn) [reply app/models/reports/platform_report.rb]: @labradford for clarification:
- can we only search 1 month at a time? 
- or is it that we have to search in the span of an entire month? (e.g. all of august **and** september, but not aug 10 - sept 29)

**Review by alishaevn — APPROVED** [review]

File: app/models/reports/platform_report.rb (hunk 1 above)
Comment (labradford) [reply app/models/reports/platform_report.rb]: any reports will be for a complete month. there will be no partial month reports. For example, if you are querying for aug 2023, the response should be "that report is not ready" or something like that

**Pull request merged**